cli/tests/test_next_prev_commands.rs

```diff
@@ -0,0 +1,223 @@
+    """###
+    );
+}
+
+#[test]
+fn test_prev_simple() {
+    // Move from third => second.
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "second"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "third"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["prev"]);
+    // The working copy commit is now a child of "second".
+    insta::assert_snapshot!(
+        stdout,
+        @r###"
+    Working copy now at: 973e9c36d3e4 (no description set)
+    Parent commit      : 3fa8931e7b89 third
+    "###
+    );
+}
+
+#[test]
+fn test_prev_multiple_without_root() {
+    // Move from fourth => second.
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "second"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "third"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "fourth"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["prev", "2"]);
+    insta::assert_snapshot!(
+        stdout,
+        @r###"
+    Working copy now at: 5647d685026f (no description set)
+    Parent commit      : 5c52832c3483 second
+    "###
+    );
+}
+
+#[test]
+fn test_next_creates_new_commit() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "second"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "third"]);
+    // Calling `next` now, should create a new commit as it's the last commit
+    // in the repositories history.
+    let stdout = test_env.jj_cmd_success(&repo_path, &["next"]);
+    insta::assert_snapshot!(stdout, @r#""#);
+}
+
+#[test]
+fn test_next_exceeding_history() {
+    //
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "second"]);
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "third"]);
+    test_env.jj_cmd_success(&repo_path, &["edit", "-r", "@--"]);
+    let stderr = test_env.jj_cmd_cli_error(&repo_path, &["next 2"]);
+    // `jj next` beyond existing history with no `new` possibility.
+    insta::assert_snapshot!(stderr, @r#""#);
```

Comment:
Seems like we can do better than an empty error message :) (if I'm reading this correctly)

Reply:
Here my wrong input to `jj_cmd_cli_error` is the reason for the missing output.

Reply:
should be fixed now